psh/cat/cat.c

```diff
@@ -36,35 +36,54 @@ int psh_cat(int argc, char **argv)
 	FILE *file;
 	char *buff;
 	size_t ret;
-	int c;
+	int c, retval = EXIT_SUCCESS;
+
+	while (1) {
+		c = getopt(argc, argv, "h");
+		if (c == -1)
+			break;
 
-	while ((c = getopt(argc, argv, "h")) != -1) {
 		switch (c) {
-		case 'h':
-		default:
-			psh_cat_help(argv[0]);
-			return EOK;
+			case 'h':
+			default:
+				psh_cat_help(argv[0]);
+				return EXIT_SUCCESS;
 		}
 	}
 
-	if ((buff = malloc(1024)) == NULL) {
+	/* FIXME? We're operating on streams, there should be no need for big buffers */
```

Comment:
comment: on nommu the file stream buffers are small (`BUFSIZ` is 16bytes) so external buffer would probably improve the performance - current implementation of `fread` always uses internal buffer for first read from underlying `fd`, then only reads the remainder directly. Probably we could disable buffering (or use our buffer for stream) using `setvbuf` for small performance improvement

Reply:
I don't know if we want to do setvbuf in an applet - only if we're able to reliably restore initial state of stdout, I don't think there is a method to get it's initial state

Reply:
Hey @Mirage1348, thanks for Your input but please stick to english language in the discussion.

If google translate understands You correctly, You're saying that we can use `setvbuf` after fork without any issue. That's true, but doesn't apply in this use case - `psh` applets are run inline without forking (to minimize the resources usage and work on NOMMU targets) :).

Reply:
Thank you very much, I am very new, what should I buy, a little for what you said 🙈